controllers/provider-openstack/pkg/apis/openstack/types_controlplane.go

```diff
@@ -0,0 +1,33 @@
+package openstack
+
+import (
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ControlPlaneConfig contains configuration settings for the control plane.
+type ControlPlaneConfig struct {
+	metav1.TypeMeta
+
+	// LoadBalancerProvider is the name of the load balancer provider in the OpenStack environment.
+	LoadBalancerProvider string
+
+	// DHCPDomain is the dhcp domain of the OpenStack system configured in nova.conf. Only meaningful for
+	// Kubernetes 1.10.1+. See https://github.com/kubernetes/kubernetes/pull/61890 for details.
+	// +optional
+	DHCPDomain *string
+	// RequestTimeout specifies the HTTP timeout against the OpenStack API.
+	// +optional
+	RequestTimeout *string
```

Comment:
The type of a timeout should not be `string`, but rather `metav1.Duration`. Kubernetes will do correct marshalling / unmarshalling to and from YAML for this type.